src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -424,22 +424,35 @@ free_unmap_bo(xclDeviceHandle handle, xclBufferHandle boh, void * boptr, size_t
  * helper function for P2P test
  */
 static bool
-p2ptest_set_or_cmp(char *boptr, size_t size, char pattern, bool set)
+p2ptest_set_or_cmp(char *boptr, size_t size, char *valid_data, size_t valid_data_size, bool set)
 {
   int stride = xrt_core::getpagesize();
 
   assert((size % stride) == 0);
+  assert(size > valid_data_size);
   for (size_t i = 0; i < size; i += stride) {
-    if (set) {
-      boptr[i] = pattern;
-    }
-    else if (boptr[i] != pattern) {
-      return false;
-    }
+    if (set)
+      std::memcpy(&(boptr[i]), valid_data, valid_data_size);
+    else if (std::memcmp(&(boptr[i]), valid_data, valid_data_size) != 0) // Verify the written bytes
```

Comment:
No need to check for not zero.
```suggestion
    else if (std::memcmp(&(boptr[i]), validData.data(), validData.size())) // Verify the written bytes
```